clang/lib/CIR/Dialect/Transforms/CIRCanonicalize.cpp

```diff
@@ -174,10 +176,11 @@ void CIRCanonicalizePass::runOnOperation() {
   // Collect operations to apply patterns.
   llvm::SmallVector<Operation *, 16> ops;
   getOperation()->walk([&](Operation *op) {
-    // CastOp and UnaryOp are here to perform a manual `fold` in
+    // CastOp, UnaryOp and VecExtractOp are here to perform a manual `fold` in
     // applyOpPatternsGreedily.
```

Comment:
@bcardosolopes do you want to keep these operations in the non-canonicalized IR?
It might be possible to directly create them using `OpBuilder::createOrFold` in codegen which will eliminate the need to fold here. But it might be worth to be able to have unfolded version too, so I am just asking.

Reply:
@xlauko good question, I wasn't aware of `createOrFold` actually, neat. I guess it really depends on optimization level given we might be folding too much if we are emitting things to be statically analyzed? I don't have a concrete example though. My answer is probably: on a case basis if it's trivial enough?